core/payment-driver/gnt/src/service.rs

```diff
@@ -182,3 +204,25 @@ async fn account_event(
 
     Ok(())
 }
+
+fn parse_network(network: Option<String>) -> Result<Network, GenericError> {
+    network
+        .unwrap_or(DRIVER_DETAILS.default_network.clone())
+        .parse()
+        .map_err(GenericError::new)
+}
+
+fn parse_platform(platform: String) -> Result<Network, GenericError> {
+    let parts: Vec<&str> = platform.split("-").collect();
```

Comment:
this stuff will not with with the metamask driver on your example, since `{drivername}-{network}-{token} !== {platform}`

Can leave it for now, maybe a note to our future selves would be nice

Reply:
This code is not meant to be re-used by other drivers. It will be scrapped once erc20 driver is refactored to use base crate.